### PR TITLE
Feature/jc 675 authorization feature angular

### PR DESCRIPTION
**Pull request by datngo-saritasa**
Create authorization function for anime app.

### CODE REVIEW

**Comment by shalomich** [comment]
Uploading bug.mp4…

**Comment by shalomich** [comment]
https://github.com/user-attachments/assets/c2971035-9f1b-44a6-a759-800319c1602c

File: apps/angular/src/app/auth/components/signup/signup.component.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+	public constructor(private formBuilder: NonNullableFormBuilder) {}
+
+	/** Sign Up Form. */
+	protected signUpForm = this.formBuilder.group({
+		email: ['', [Validators.required, Validators.email]],
+		firstName: ['', [Validators.required, Validators.maxLength(30)]],
+		lastName: ['', [Validators.required, Validators.maxLength(30)]],
+		passwordGroup: this.formBuilder.group({
+			password: ['', [Validators.required, Validators.minLength(8)]],
```
Comment (shalomich) [inline apps/angular/src/app/auth/components/signup/signup.component.ts:45]: Please have constant for minLength and to reuse it in `signin.component`

File: apps/angular/src/app/auth/components/signup/signup.component.ts [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+			reTypePassword: ['', [Validators.required, Validators.minLength(8)]],
```
Comment (shalomich) [inline apps/angular/src/app/auth/components/signup/signup.component.ts:46]: What if `retypePassword` doesn't match `password`?

**Review by shalomich — CHANGES_REQUESTED** [review]

**Comment by shalomich** [comment]
@datngo-saritasa I cannot view files unrelated to authorization, because I suspect they may not be relevant from other PR reviews. Therefore, before you ask me again for this PR, I ask you to merge other PRs into the dev and update your branch regarding it

**Comment by shalomich** [comment]
@datngo-saritasa Please fix linting and SonarCloud error

File: apps/angular/src/app/auth/auth.component.html [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+		<section class="main__background-image">
+		</section>
+		<section class="main__login">
+			<section class="main__login-wrapper">
+				<div class="main__logo">
+					<img src="../../assets/icon.png" alt="Logo">
+				</div>
+				<router-outlet></router-outlet>
+			</section>
+			<section class="main__copyright">
```
[the remark this answers is not in the document]
Comment (datngo-saritasa) [reply apps/angular/src/app/auth/auth.component.html:12]: I have read through the docs so `<section>`  is refers to a standalone group of content that do not have any sematic tag fit to its but have to have a header. So in this case if we did not have any specific tag and cannot use `<section>`, we can use `div` like a division of content.

File: apps/angular/src/app/auth/components/password-input/password-input.component.css [hunk 4]
```diff
@@ -0,0 +1,14 @@
+.password-form {
+	width: 100%;
+}
+
+.password-form__label {
+	color: var(--anime-japan-title-row-color);
```
[the remark this answers is not in the document]
Comment (datngo-saritasa) [reply apps/angular/src/app/auth/components/password-input/password-input.component.css:6]: For this I just reuse the color for those 2 field I will make another name for the color variable.

File: apps/angular/src/app/auth/components/signin/signin.component.css [hunk 5]
```diff
@@ -0,0 +1,61 @@
+.form__input-group {
+	margin-bottom: var(--spacing-md);
+	display: grid;
+}
+
+.form__label {
+	color: var(--anime-button);
```
[the remark this answers is not in the document]
Comment (datngo-saritasa) [reply apps/angular/src/app/auth/components/signin/signin.component.css:7]: Same problems I reuse the color but not have a general name so I will redo the color name.

File: apps/angular/src/app/auth/components/signin/signin.component.html [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+					placeholder="Enter password"
+					[touched]="signInForm.controls.password.touched"
+				></camp-password-input>
+	</div>
+	<section class="form__option">
+		<div class="form__remember-me">
+			<input type="checkbox" id="rememberMe" />
+			<label for="rememberMe">Remember me</label>
+		</div>
+		<a href="#" class="form__forgot-password">Forgot password?</a>
```
[the remark this answers is not in the document]
Comment (datngo-saritasa) [reply apps/angular/src/app/auth/components/signin/signin.component.html:20]: I will remove the `href` link but leave the `<a>` element for later use.

File: apps/angular/src/app/auth/components/signup/signup.component.ts (hunk 2 above)
Comment (datngo-saritasa) [reply apps/angular/src/app/auth/components/signup/signup.component.ts:46]: Sorry I have develop a mustMatch Fn in `apps\angular\src\core\utils\custom-validator.ts` but havent use it. Thank for remind me.

File: apps/angular/src/app/auth/components/signin/signin.component.html [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+		></camp-password-input>
+	</div>
+	<section class="form__option">
+		<div class="form__remember-me">
+			<input type="checkbox" id="rememberMe" />
+			<label for="rememberMe">Remember me</label>
+		</div>
+		<a class="form__forgot-password">Forgot password?</a>
+	</section>
+	<button type="submit" class="form__btn-signin" [disabled]="signInForm.invalid && (isLoading$ | async)">
```
Comment (daoduong-saritasa) [inline apps/angular/src/app/auth/components/signin/signin.component.html:22]: This should be `||` , `isLoading$` can't be triggered if form is invalid

File: apps/angular/src/app/auth/components/signin/signin.component.ts [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+	PASSWORD_MAX_LENGTH,
+	EMAIL_MIN_LENGTH,
+	EMAIL_MAX_LENGTH,
+} from '@js-camp/angular/shared/constant';
+
+import { errorGuard } from '@js-camp/angular/core/guards/error-guard';
+
+import { PasswordInputComponent } from '../password-input/password-input.component';
+
+/** Signin. */
```
Comment (daoduong-saritasa) [inline apps/angular/src/app/auth/components/signin/signin.component.ts:26]:
```suggestion
/** Signin component. */
```

File: apps/angular/src/app/auth/components/signin/signin.component.ts [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+export class SignInComponent {
+	private readonly userService = inject(UserService);
+
+	private readonly formErrorService = inject(FormErrorService);
+
+	private readonly router = inject(Router);
+
+	private readonly notificationService = inject(NotificationService);
+
+	private formBuilder = inject(NonNullableFormBuilder);
```
Comment (daoduong-saritasa) [inline apps/angular/src/app/auth/components/signin/signin.component.ts:52]:
```suggestion
	private readonly formBuilder = inject(NonNullableFormBuilder);
```

File: apps/angular/src/app/auth/components/signin/signin.component.ts [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+	protected readonly signInForm = this.formBuilder.group({
+		email: [
+			'',
+			[
+				Validators.required,
+				Validators.email,
+				Validators.minLength(EMAIL_MIN_LENGTH),
+				Validators.maxLength(EMAIL_MAX_LENGTH),
+			],
+		],
```
Comment (daoduong-saritasa) [inline apps/angular/src/app/auth/components/signin/signin.component.ts:70]:
```suggestion
		email: this.formBuilder.control('', [
				Validators.required,
				Validators.email,
				Validators.minLength(EMAIL_MIN_LENGTH),
				Validators.maxLength(EMAIL_MAX_LENGTH),
			]),
```

File: apps/angular/src/app/auth/components/signin/signin.component.ts [hunk 11]
```diff
@@ -0,0 +1,125 @@
...
+		password: [
+			'',
+			[Validators.required, Validators.minLength(PASSWORD_MIN_LENGTH), Validators.maxLength(PASSWORD_MAX_LENGTH)],
+		],
+	});
+
+	/** Display error.
+	 * @param controlName Name of the control.
```
Comment (daoduong-saritasa) [inline apps/angular/src/app/auth/components/signin/signin.component.ts:78]:
```suggestion
	/** 
	 * Display error.
	 * @param controlName Name of the control.
```

File: apps/angular/src/app/auth/components/signin/signin.component.ts [hunk 12]
```diff
@@ -0,0 +1,125 @@
...
+	 * @returns Should show error of control field.
+	 */
+	protected shouldShowError(controlName: string): boolean {
+		const formControl = this.signInForm.get(controlName) as FormControl;
+		return this.formErrorService.shouldShowError(formControl);
+	}
+
+	/**
+	 *	Get Error message.
```
Comment (daoduong-saritasa) [inline apps/angular/src/app/auth/components/signin/signin.component.ts:87]: Wrong indentation

File: apps/angular/src/app/auth/components/signup/signup.component.ts [hunk 13]
```diff
@@ -0,0 +1,141 @@
...
+	changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class SignupComponent {
+	private readonly formErrorService = inject(FormErrorService);
+
+	private readonly authService = inject(UserService);
+
+	private readonly router = inject(Router);
+
+	private formBuilder = inject(NonNullableFormBuilder);
```
Comment (daoduong-saritasa) [inline apps/angular/src/app/auth/components/signup/signup.component.ts:42]:
```suggestion
	private readonly formBuilder = inject(NonNullableFormBuilder);
```

File: apps/angular/src/app/auth/components/signup/signup.component.ts [hunk 14]
```diff
@@ -0,0 +1,141 @@
...
+
+	private readonly destroyRef = inject(DestroyRef);
+
+	/** Form errors. */
+	protected formErrors: { [key: string]: string | null; } = {};
+
+	/** Sign Up Form. */
+	protected signUpForm = this.formBuilder.group({
+		email: [
```
Comment (daoduong-saritasa) [inline apps/angular/src/app/auth/components/signup/signup.component.ts:51]: Please update form control with `this.formBuilder.control`

File: apps/angular/src/app/auth/components/signup/signup.component.ts [hunk 15]
```diff
@@ -0,0 +1,141 @@
...
+		if (data == null) {
+			return null;
+		}
+		return this.formErrorService.getErrorMessage(data);
+	}
+
+	/** Submit form. */
+	protected onSubmit(): void {
+		this.signUpForm.markAllAsTouched();
+		if (this.signUpForm.valid) {
```
Comment (daoduong-saritasa) [inline apps/angular/src/app/auth/components/signup/signup.component.ts:105]: Please use early return with `this.signUpForm.invalid`

File: apps/angular/src/app/features/anime/anime-catalog/anime-catalog.component.ts [hunk 16]
```diff
@@ -0,0 +1,159 @@
...
+
+	/** Sort params. */
+	protected readonly sortParams$ = new BehaviorSubject<Sort>({
+		direction: '',
+		active: '',
+	});
+
+	public constructor() {
+		this.animePage$ = this.filter$.pipe(
+			debounceTime(1000),
```
Comment (daoduong-saritasa) [inline apps/angular/src/app/features/anime/anime-catalog/anime-catalog.component.ts:77]: Avoid to use magic number, declare a variable instead

File: apps/angular/src/app/features/anime/anime-catalog/anime-catalog.component.ts [hunk 17]
```diff
@@ -0,0 +1,159 @@
...
+				this.animeService.getAnime(queryParams).pipe(
+					finalize(() => {
+						this.isLoading$.next(false);
+					}),
+				)),
+		);
+		this.currentUser$ = this.userService.currentUser$;
+	}
+
+	/** Subscribe the filter params and pass them to the filter form and paginator. */
```
Comment (daoduong-saritasa) [inline apps/angular/src/app/features/anime/anime-catalog/anime-catalog.component.ts:91]: Use `/** @inheritdoc */` for life-cycle functions, the name of the side effect should explain what it does, not life-cycle docs

**Review by daoduong-saritasa — CHANGES_REQUESTED** [review]
This branch should be merge into `feature/JC-656-anime-list-management` not `develop`. Seem like you checkout from `JC-656` and it's not merge to develop yet, so this branch will contain both `JC-656` and `JC-675` commits when merge to develop.

File: apps/angular/src/app/auth/components/signin/signin.component.html [hunk 18]
```diff
@@ -0,0 +1,28 @@
+<form [formGroup]="signInForm" (ngSubmit)="onSubmit()">
+	<div class="form__input-group">
```
[the remark this answers is not in the document]
Comment (datngo-saritasa) [reply apps/angular/src/app/auth/components/signin/signin.component.html:2]: I have add the form class for the form.

**Pull request closed without merging**